Mathlib/Algebra/Notation/Pi.lean

```diff
@@ -13,56 +14,135 @@ This file provides only the notation for (pointwise) `0`, `1`, `+`, `*`, `•`,
 See `Mathlib.Algebra.Group.Pi.Basic` for the `Monoid` and `Group` instances.
 -/
 
-assert_not_exists Set.range Monoid DenselyOrdered
+assert_not_exists Monoid Preorder
 
 open Function
 
 universe u v₁ v₂ v₃
 
-variable {I : Type u}
+variable {ι α β γ : Type*} variable {M N O : ι → Type*} (x y : ∀ i, M i) (i : ι)
 
--- The indexing type
-variable {α β γ : Type*}
+namespace Pi
 
--- The families of types already equipped with instances
-variable {f : I → Type v₁} (x y : ∀ i, f i) (i : I)
+-- TODO: Do we really need this definition? If so, where to put it?
```

Comment:
I think I added it to match `LinearMap.prod` etc.

Reply:
Aha, but were you the one to put it in a seemingly random file?